auto_round/utils/device.py

```diff
@@ -415,6 +415,62 @@ def clear_memory(tensor=None):
         _clear_memory_for_cpu_and_cuda(tensor)
 
 
+def clear_memory_if_reached_threshold(threshold=0.85):
+    """Check all available devices and clear memory if any device is using close to the threshold.
+
+    Args:
+        threshold (float): Memory usage threshold (default: 0.85 for 85%).
+                            If any device exceeds this percentage, clear_memory() will be called.
+
+    Returns:
+        bool: True if memory was cleared, False otherwise.
+    """
+    # Check CUDA devices
+    if torch.cuda.is_available():
+        num_devices = torch.cuda.device_count()
+        for i in range(num_devices):
+            try:
+                total_memory = torch.cuda.get_device_properties(i).total_memory
```

Comment:
Would it be better to wrap some functions to make memory information retrieval device-agnostic